pkg/restapi/operation/operations.go

```diff
@@ -124,23 +136,25 @@ Request body: %s`, string(requestBody))
 		}
 	}
 
-	c.createDataVault(rw, &config, req.Host, configBytesForLog)
+	c.createDataVault(rw, &config, req.Host, configBytesForLog, &uuidGenerator{})
 }
 
 func (c *Operation) createDataVault(rw http.ResponseWriter, config *models.DataVaultConfiguration, hostURL string,
-	configBytesForLog []byte) {
+	configBytesForLog []byte, uuidBuilder UUIDBuilder) {
```

Comment:
You need to make sure that two vaults can't have the same referenceID, similar to the Transmute implementation... this means you'll need to store that DataVaultConfiguration somewhere...